python/cudf/cudf/core/index.py

```diff
@@ -1641,10 +1832,106 @@ def as_index(arbitrary, **kwargs):
     )
 
 
-def _setdefault_name(values, kwargs):
-    if "name" not in kwargs:
+def _setdefault_name(values, **kwargs):
+    if "name" not in kwargs or kwargs["name"] is None:
         if not hasattr(values, "name"):
-            kwargs.setdefault("name", None)
+            kwargs.update({"name": None})
         else:
-            kwargs.setdefault("name", values.name)
+            kwargs.update({"name": values.name})
     return kwargs
+
+
+def Int64Index(data=None, dtype="int64", copy=False, name=None):
```

Comment:
Maybe I'm missing something, but these need to be actual types, not functions.  Someone should be able to do `type(x) is cudf.Int64Index == True` for instance.